src/schedule/mod.rs

```diff
@@ -76,7 +85,7 @@ impl Default for EpochSecrets {
     fn default() -> Self {
         let welcome_secret = Secret::new_empty_secret();
         let sender_data_secret = Secret::new_empty_secret();
-        let encryption_secret = Secret::new_empty_secret();
+        let encryption_secret = RefCell::new(Some(Secret::new_empty_secret()));
```

Comment:
Is there a reason you need the `RefCell` here? Why not just an `Option` that you can later consume with`.take()`?

Reply:
Good point. That means we need to make `EpochSecrets` `mut`, but that should be just fine.